src/Memory.re

```diff
@@ -37,11 +37,23 @@ let get_byte = (mem: t, loc: address): int =>
     (mem.mapper)#get_prg(loc);
   };
 
+let dma = (mem, value) => {
+  let page = value lsl 8;
+  for (i in 0 to 255) {
+    let index = page + i;
+    let oam_byte = get_byte(mem, index);
+    Ppu.store(mem.ppu, 4, oam_byte);
+  };
+  // TODO: Implement CPU wait cycles so PPU doesn't spend a few scanlines "catching up".
```

Comment:
We should discuss a nice way for the Memory to specially adjust CPU cycle count here. The CPU can't pass it to memory at construction because it needs Memory for its own construction. It would be quite silly to always pass a closure to `set_byte` and placing `dma` in the Cpu or Nes module results in a cyclic dependency.

See: https://wiki.nesdev.com/w/index.php/PPU_programmer_reference#OAM_DMA_.28.244014.29_.3E_write